readthedocs/config/config.py

```diff
@@ -264,11 +264,11 @@ def python_interpreter(self):
     @property
     def python_full_version(self):
         ver = self.python.version
-        if ver in [2, 3]:
+        if ver in [2, 3, '2', '3']:
```

Comment:
Think we can just transform this to a string instead of handling strings and numbers, `ver = str(self.python.version) if self.python.version is not None else None`

Reply:
(this can be done when populating python.version instead of here is seems)

Reply:
> (this can be done when populating python.version instead of here is seems)

Here you mean by casting it as `str()`? https://github.com/readthedocs/readthedocs.org/blob/80b3c2c5b8bc06ea312efe16830e1753dfa4b856/readthedocs/config/config.py#L516-L519

If so, I'm not 100% convinced if it's good to change the type written by the user automatically --that problem led us here. I think we will face other issues silently. Are there other benefits besides the sugar syntax of this line?

Reply:
> Here you mean by casting it as str()? 

Yes

> I think we will face other issues silently. Are there other benefits besides the sugar syntax of this line?

We don't  need to take into consideration strings and numbers in all our code base, don't think this will cause problems, if you mean the users setting this value as `3.50`, that will be cast as `3.5` already (float), transforming that to string leads to the same `3.5`

Reply:
I'm 👍🏼 on always using string internally. However, changing this to always use string internally requires some extra work. There are a bunch of tests to adapt and some internal functions to change and maybe re-think it's logic. I may consider this a refactor outside the scope of this PR and we can do this work in another one, probably.

If we want to support 3.10 and newer conda starting tomorrow after the deploy, I think we should merge this solution.

Reply:
Also, I'd appreciate it if you have the time to jump into this today so we can deploy it tomorrow with the appropriate changes.

Reply:
I don't think we should rush a big change like this just to get it working by an arbitrary deploy. Let's take the time to properly design the fix, and push the deploy a week if needed. We're only using the config file internally, so we should be able to make changes to it without too much issue.

I think the primary issue I'm expecting is a user who passes `3.10` as a float and it gets read as `3.1`. What is our plan here -- automatically cast floats of 3.1 to strings of `3.10`? Fail the build if the user passes `3.10` instead of `'3.10'`?

Reply:
> I think the primary issue I'm expecting is a user who passes 3.10 as a float and it gets read as 3.1

That's a good question, looks like the yaml spec automatically cast those, so not much we can do, we can change all of our examples to always use strings

```python
>>> yaml.safe_load(Path('test.yaml').open())
{'version': 3.1, 'version2': '3.10'}
```

```yaml
version: 3.10
version2: "3.10"
```

Reply:
> Fail the build if the user passes 3.10 instead of '3.10'?

I think that should be the case, and we should always use strings in our docs and maybe put a warning instead of magically changing 3.1 to 3.10

Reply:
Yea, from the discussion, it seems like using floats for this is broken by design, and we need to systemically move to strings everywhere. I think we want to maintain backwards compat here though, so we should continue to accept the strings we have historically, but not new ones. So, for example `3.10` would error as a float, with a very good error message -- but I'm not 100% sure that's correct. It will be really annoying to users, since we _know_ what they want, and could trivially convert it. 

I'm torn on the correct path forward here, but I'm definitely convinced we should move to strings everywhere, and the question is how to handle the immediate 3.10 support and backwards compat. If we have a strong need to ship this code this week, I'm OK with special-casing `'3.10'` for now since we can make it less strict over time, but not more.  We should pretty quickly follow up with a larger "strings everywhere" fix and docs update though I think.

Reply:
Thinking more on this, I think we need to convert `3.10` to the proper value instead of breaking. Imagine a user upgrading from `3.9` to `3.10` -- if they just changed their version in the config, it would blow up -- this seems like a terrible UX to have the same field map to various data types depending on value. We should at least support strings for _all_ versions, if we're going to make this upgrade.

Reply:
I have opened https://github.com/readthedocs/readthedocs.org/pull/8372 changing everything to strings (but numbers still work), and with a special case for 3.10 (number)